src/Microsoft.AspNetCore.Authentication/RemoteAuthenticationHandler.cs

```diff
@@ -89,9 +89,27 @@ public abstract class RemoteAuthenticationHandler<TOptions> : AuthenticationHand
 
         protected abstract Task<AuthenticateResult> HandleRemoteAuthenticateAsync();
 
-        protected override Task<AuthenticateResult> HandleAuthenticateAsync()
+        protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
         {
-            return Task.FromResult(AuthenticateResult.Fail("Remote authentication does not support authenticate"));
+            var authenticateContext = new AuthenticateContext(Options.SignInScheme);
+            await PriorHandler.AuthenticateAsync(authenticateContext);
+            if (authenticateContext.Accepted)
+            {
+                if (authenticateContext.Error != null)
+                {
+                    return AuthenticateResult.Fail(authenticateContext.Error);
+                }
+                // Compare the ClaimsIssuer to make sure the identity was produced by this provider and not by some other one.
+                if (string.Equals(authenticateContext.Principal?.Identity?.AuthenticationType, Options.ClaimsIssuer, StringComparison.Ordinal))
```

Comment:
I think it would be very strange for `[Authorize("Github")]` to return a Facebook identity. I'm imagining a scenario where you need to sign in with a specific provider to access specific functionality like a github API call.  I'll update the Social Sample. Of course this approach doesn't work when you're using Identity as it has its own auth types.

ClaimsIssuer is used to set the ClaimsIdentity AuthType in all the handlers, and it's set to the middleware AuthType by default.